File: programmers/js/basic/181851.js
// 전국 대회 선발 고사
function solution(rank, attendance) {
  const oneTwoThree = [];
  const sortOfRank = [
    ...rank.reduce(
      (acc, cur, idx) => acc.set(cur, [idx, attendance[idx]]),
      new Map()
    ),
  ].sort((a, b) => a[0] - b[0]);

  for (const [rnk, [num, atd]] of sortOfRank) {
    if (oneTwoThree.length > 3) break;

    if (atd) {
      oneTwoThree.push(num);
    }
  }

  return 10000 * oneTwoThree[0] + 100 * oneTwoThree[1] + oneTwoThree[2];
}

File: programmers/java/basic/181851.java
// 전국 대회 선발 고사

import java.util.HashMap;
import java.util.ArrayList;


class Solution {
    public int solution(int[] rank, boolean[] attendance) {
        HashMap<String, Integer> map = new HashMap<>();

        for (int i = 0; i < rank.length; i++) {
            if (attendance[i]) {
                map.put(Integer.toString(rank[i]), i);
            }
        }

        ArrayList<String> keyList = new ArrayList<>(map.keySet());
        keyList.sort((k1, k2) -> Integer.parseInt(k1) - Integer.parseInt(k2));
        int result = 0;

        return 10000 * map.get(keyList.get(0)) + 
            100 * map.get(keyList.get(1)) +
            map.get(keyList.get(2));
    }
}